ratik/src/main/java/org/opencadc/ratik/ArtifactValidator.java

```diff
@@ -626,6 +651,32 @@ Artifact getRemoteArtifact(URI uri)
         return null;
     }
 
+    Integer getRemoteArtifactCount(URI uri)
```

Comment:
This has to be a query by Artifact.id -- if the local artifact was recently PUT here it could have a different ID than the old (overwritten) entity in global... the num_copies() of the local artifact could still be 0/null/not-found (fenwick not yet propagated) while the old one might have 2+ copies. That would trigger a delete of the wrong (newer) artifact. 

context: in the design, a site with an artifact-filter that excludes an artifact can still end up accepting a put of such a file (eg acting as a cache if writes to other "preferred" sites fail)... that's how one could get an "unwanted" local artifact but have to wait until "that artifact" (not an artifact with the same uri) is propagated and the delete is safe.

Reply:
Okay I see the difference, updated to query using the ID.